### PR TITLE
MRG: Fix covariance order

### PR DESCRIPTION
**Pull request by Eric89GXL**
Fixes a bug where the channel order / indexing in `prepare_noise_cov` could lead to `AssertionError` or (worse) silent errant selection of channels.

### CODE REVIEW

File: mne/cov.py [hunk 1]
```diff
@@ -1275,17 +1282,33 @@ def prepare_noise_cov(noise_cov, info, ch_names, rank=None,
...
+        info_pick_meg = pick_channels(info['ch_names'], meg_names)
+    else:
+        info_pick_meg = []
+    if len(eeg_names) > 0:
+        info_pick_eeg = pick_channels(info['ch_names'], eeg_names)
+    else:
+        info_pick_eeg = []
+    assert len(info_pick_meg) == len(meg_names) == len(out_meg_idx)
+    assert len(info_pick_eeg) == len(eeg_names) == len(out_eeg_idx)
+    assert(len(out_meg_idx) + len(out_eeg_idx) == n_chan)
```
Comment (agramfort) [inline mne/cov.py]: use assert_true

Comment (agramfort) [inline mne/cov.py]: forget it I thought it was a test

**Comment by agramfort** [comment]
LGTM

any other review for this important code?

**Comment by kambysese** [comment]
Worked on my end. +1 for merge. Thanks @Eric89GXL

**Comment by agramfort** [comment]
thx @Eric89GXL and @kambysese

**Pull request merged**